src/services/hive/hive.service.ts

```diff
@@ -69,21 +77,20 @@ export class HiveService {
       );
     }
 
-    const accountCreation = await this.#createAccountWithAuthority(hiveUsername);
-
-    console.log(accountCreation);
+    const accountCreation = await this.#createAccountWithAuthority(hiveUsername, sub);
 
     await this.insertCreated(hiveUsername, sub);
 
     return accountCreation;
   }
 
-  async #createAccountWithAuthority(hiveUsername: string) {
+  async #createAccountWithAuthority(hiveUsername: string, sub: string) {
     if (!process.env.ACCOUNT_CREATOR) {
       throw new Error('Please set the ACCOUNT_CREATOR env var');
     }
     try {
-      return await this.#hiveRepository.createAccountWithAuthority(
+      await this.#linkedAccountsRepository.linkHiveAccount(sub, hiveUsername);
+      return await this.#hiveChainRepository.createAccountWithAuthority(
         hiveUsername,
         process.env.ACCOUNT_CREATOR,
       );
```

Comment:
These operations need to be flipped in case that the account creation fails. We don't want to give anyone easy access to linking with some else's account